stopes/pipelines/asr_bleu/transcribe_audio.py

```diff
@@ -0,0 +1,213 @@
+
+    def _decode_emissions(self,emissions: torch.Tensor, decoder, post_process_fn) -> str:
+        """
+        Decode the emissions and apply post-process functions
+
+        Args:
+            emissions: the input Tensor object
+            decoder: ASR decoder
+            post_process_fn: post-process function for the ASR output
+
+        Returns:
+            hypo: the str as the decoded transcriptions
+        """
+
+        emissions = emissions.cpu()
+        results = decoder(emissions)
+
+        # assuming the lexicon-free decoder and working with tokens
+        hypo = decoder.idxs_to_tokens(results[0][0].tokens)
+        hypo = post_process_fn(hypo)
+
+        return hypo
+
+    def _merge_tailo_init_final(self, text):
+        """
+        Hokkien ASR hypothesis post-processing.
+        """
+        sps = text.strip().split()
+        results = []
+        last_syllable = ""
+        for sp in sps:
+            if sp == "NULLINIT" or sp == "nullinit":
+                continue
+            last_syllable += sp
+            if sp[-1].isnumeric():
+                results.append(last_syllable)
+                last_syllable = ""
+        if last_syllable != "":
+            results.append(last_syllable)
+        return " ".join(results)
+
+    def _remove_tone(self, text):
+        """
+        Used for tone-less evaluation of Hokkien
+        """
+        return " ".join([t[:-1] for t in text.split()])
+
+    def _transcribe_audiofile(
+        self,  
+        asr_model,
+        audio_path: str, 
+        lower=True
+    ) -> str:
+        """
+        Transcribe the audio into a string
+
+        Args:
+            model_cfg: the dict of the asr model config
+            audio_path: the input audio waveform
+            lower: the case of the transcriptions with lowercase as the default
+
+        Returns:
+            hypo: the transcription result
+        """
+
+        asr_input = self._load_audiofile(audio_path, asr_model.sampling_rate, asr_model.normalize_input)
+        emissions = self._compute_emissions(asr_input, asr_model.model, asr_model.use_cuda)
+        hypo = self._decode_emissions(emissions, asr_model.decoder, asr_model.post_process_fn)
+
+        return hypo.strip().lower() if lower else hypo.strip()
+
+    def run(self, 
+            iteration_value: tp.Optional[tp.Any] = None,
+            iteration_index: int = 0,
+    ) -> tp.List[str]:
+        
+        assert iteration_value is not None, "Iteration value is null"
+        self.logger = logging.getLogger("stopes.asr_bleu.transcribe_audio")
+        asr_model = ASRGenerator(iteration_value.asr_config)
+
+        prediction_transcripts = []
+    
+        #TODO: reimplement tqdm progress bar with the dictionary instead of the dataframe
+        #for _,  eval_pair in tqdm(
+        #    iteration_value.eval_manifest.iterrows(),
+        #    desc="Transcribing predictions",
+        #    total=len(iteration_value.eval_manifest["prediction"]),
+        #):
```

Comment:
Make sure you exclude the commented out lines in your PR.

Reply:
Thanks, resolved in https://github.com/facebookresearch/stopes/pull/52/commits/c6e830f1225545544f9d104b27874c5479925491